src/dscanner/analysis/local_imports.d

```diff
@@ -5,99 +5,99 @@
 
-	/**
-	 * Construct with the given file name.
-	 */
-	this(string fileName, const(Scope)* sc, bool skipTests = false)
+	mixin ScopedVisit!(AST.FuncDeclaration);
+	mixin ScopedVisit!(AST.IfStatement);
+	mixin ScopedVisit!(AST.WhileStatement);
+	mixin ScopedVisit!(AST.ForStatement);
+	mixin ScopedVisit!(AST.ForeachStatement);
+	mixin ScopedVisit!(AST.ClassDeclaration);
+	mixin ScopedVisit!(AST.StructDeclaration);
+	
+	extern(D) this(string fileName)
 	{
-		super(fileName, sc, skipTests);
+		super(fileName);
+		this.localImport = false;
 	}
 
-	mixin visitThing!StructBody;
-	mixin visitThing!BlockStatement;
-
-	override void visit(const Declaration dec)
+	override void visit(AST.Import i)
 	{
-		if (dec.importDeclaration is null)
-		{
-			dec.accept(this);
-			return;
-		}
-		foreach (attr; dec.attributes)
+		// Look for import foo.bar : x or foo.bar : y = x
+		if (!i.isstatic && localImport && i.names.length == 0 && !i.aliasId)
 		{
-			if (attr.attribute == tok!"static")
-				isStatic = true;
+			addErrorMessage(cast(ulong) i.loc.linnum, cast(ulong) i.loc.charnum, KEY, MESSAGE);
 		}
-		dec.accept(this);
-		isStatic = false;
 	}
 
-	override void visit(const ImportDeclaration id)
+	// Skip unittests for now
+	override void visit(AST.UnitTestDeclaration ud)
 	{
-		if ((!isStatic && interesting) && (id.importBindings is null
-				|| id.importBindings.importBinds.length == 0))
-		{
-			foreach (singleImport; id.singleImports)
-			{
-				if (singleImport.rename.text.length == 0)
-				{
-					addErrorMessage(singleImport.identifierChain.identifiers[0].line,
-							singleImport.identifierChain.identifiers[0].column,
-							"dscanner.suspicious.local_imports", "Local imports should specify"
-							~ " the symbols being imported to avoid hiding local symbols.");
-				}
-			}
-		}
+		return;
 	}
 
 private:
-
-	mixin template visitThing(T)
+	template ScopedVisit(NodeType)
 	{
-		override void visit(const T thing)
+		override void visit(NodeType n)
 		{
-			const b = interesting;
-			interesting = true;
-			thing.accept(this);
-			interesting = b;
+			bool prevState = localImport;
+			localImport = true;
+			super.visit(n);
+			localImport = prevState;
 		}
 	}
 
-	bool interesting;
-	bool isStatic;
+	bool localImport;
+	enum KEY = "dscanner.suspicious.local_imports";
+	enum MESSAGE = "Local imports should specify the symbols being imported to avoid hiding local symbols.";
 }
 
 unittest
 {
 	import dscanner.analysis.config : StaticAnalysisConfig, Check, disabledConfig;
+	import std.stdio : stderr;
 
 	StaticAnalysisConfig sac = disabledConfig();
 	sac.local_import_check = Check.enabled;
-	assertAnalyzerWarnings(q{
-		void testLocalImport()
+
+	assertAnalyzerWarningsDMD(q{
+		import std.experimental;
+
+		void foo()
 		{
 			import std.stdio; // [warn]: Local imports should specify the symbols being imported to avoid hiding local symbols.
 			import std.fish : scales, head;
 			import DAGRON = std.experimental.dragon;
+
+			if (1) {
+
+			} else {
```

Comment:
Please be consistent with the style which is curly braces on their own line. Also, why not test the if branch also?